eessi/testsuite/tests/apps/osu.py

```diff
@@ -0,0 +1,296 @@
+"""
+This module tests the binary 'osu' in available modules containing substring 'OSU-Micro-Benchmarks'.
+The basic application class is taken from the hpctestlib to which extra features are added.
+"""
+
+import os
+import reframe as rfm
+from reframe.core.meta import parameters
+import reframe.utility.sanity as sn
+
+from hpctestlib.microbenchmarks.mpi.osu import osu_benchmark
+
+from eessi.testsuite import hooks, utils
+from eessi.testsuite.constants import SCALES, TAGS, DEVICE_TYPES
+
+def my_filtering_function():
+    """
+    Filtering function for filtering scales for the pt2pt OSU test
+    """
+    scale_filtered = SCALES.copy()
+    for key in list(SCALES):
+        if(key == '1_core' or key == '4_cores' or
+           SCALES.get(key).get('num_nodes') > 2):
+            scale_filtered.pop(key)
+        elif('node_part' in SCALES.get(key)):
+            if(SCALES.get(key).get('node_part') > 1):
+                scale_filtered.pop(key)
+    return scale_filtered
+
+
+def my_filtering_function_coll():
+    """
+    Filtering function for filtering scales for collective the OSU test
+    """
+    scale_filtered = SCALES.copy()
+    for key in list(SCALES):
+        if(key == '1_core'):
+            scale_filtered.pop(key)
+    return scale_filtered
+
+
+@rfm.simple_test
+class osu_pt_2_pt(osu_benchmark):
+    ''' Run-only OSU test '''
+    scale = parameter(my_filtering_function())
+    valid_prog_environs = ['default']
+    valid_systems = []
+    time_limit = '30m'
+    module_name = parameter(utils.find_modules('OSU-Micro-Benchmarks'))
+    # Device type for non-cuda OSU-Micro-Benchmarks should run on hosts of both
+    # node types. To do this the default device type is set to GPU.
+    device_type = DEVICE_TYPES['GPU']
```

Comment:
i would make this a parameter so we can run both on cpus and gpus
```suggestion
    device_type = parameter([DEVICE_TYPES['CPU'], DEVICE_TYPES['GPU'])
```

Reply:
This one I agree with and have incorporated this.